src/packageurl/__init__.py

```diff
@@ -530,7 +530,10 @@ def from_string(cls, purl: str) -> "PackageURL":
         ns_name_parts = ns_name.split("/")
         ns_name_parts = [seg for seg in ns_name_parts if seg and seg.strip()]
         name = ""
-        if not namespace and len(ns_name_parts) > 1:
+        if type.lower() == "golang":
```

Comment:
I really do not like these warts for each type: npm, now golang .... *sigh* ... eventually we will need to bubble these issue up in the spec. They have little impact since they do not change the purl, only how a purl is interpreted.... But for golang there is the issue of legacy that uses a subpath.